portality/api/current/data_objects/article.py

```diff
@@ -173,7 +173,7 @@ def _remove_from_the_list_if_empty_data(bibjson_element, field=None):
             if bibjson_element in bibjson:
                 for i in range(len(bibjson[bibjson_element]) - 1, -1, -1):
                     ide = bibjson[bibjson_element][i]
-                    if field is not None:
+                    if field is not None and field in ide:
```

Comment:
This gets rid of the exception, but does it have the correct result?

It looks like the objective of the function is to remove an object OR string (represented by `ide`) from the list `bibjson_element`

The old code got the list entry `ide` and then if a field was passed to the function went on to remove that that entry from the list if the value of the field inside that entry is empty.

Now it also only tries to remove that entry if the field is also present in the list element.

But isn't the empty string as the value, and "not present", effectively equivalent, given the function is called `_remove_from_the_list_if_empty_data`?

This update would mean that a list entry would NOT be removed from the list if the key wasn't present, but the object still was.

So, this would get cleaned up

```
[
    { "author": {"name": ""}}
]
```

But this would not

```
[
    {"author": {}}
]
```

I think a better way to do this would be to continue to use the if/else as before and instead:

```
if ide.get(field, "") == ""
```

?

Reply:
Yep, you're right. It could also be changed to:
```
if field is not None:
     if field not in ide or ide[field] == "":
```